### ISSUE TITLE
ChatGLM2训练需要多大显存

### ISSUE THREAD

**Issue by fade-color** [hiyouga/ChatGLM-Efficient-Tuning#185]
使用int4量化lora微调，24G的3090会OOM，想问一下运行起来大概需要多少显存呢
训练的参数如下：
```bash
CUDA_VISIBLE_DEVICES=1 python src/train_sft.py \
    --model_name_or_path THUDM/chatglm2-6b \
    --do_train \
    --dataset code_helper \
    --finetuning_type lora \
    --lora_rank 8 \
    --quantization_bit 4 \
    --output_dir code_helper \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 8 \
    --lr_scheduler_type cosine \
    --logging_steps 10 \
    --save_steps 50 \
    --learning_rate 5e-5 \
    --num_train_epochs 4.0 \
    --fp16 \
    --use_v2
```

**Comment by xufan-zhai** [comment]
类似的问题 32g显存同样oom

**Comment by hiyouga** [comment]
不用 int4 量化呢？

**Comment by fade-color** [comment]
同样也会OOM

**Comment by testmurk** [comment]
我测试的大概需要38g batch_size 2

**Comment by hiyouga** [comment]
目前 ChatGLM2 已经实现 gradient checkpointing 并可降低显存占用，请更新代码。
